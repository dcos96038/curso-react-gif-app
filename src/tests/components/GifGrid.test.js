import { GifGrid } from "../../components/GifGrid";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en el componente GifGrid", () => {
  test("El componente debe renderizar correctamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category="messi" />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar la misma cantidad de componentes GifGridItem que gifs obtenidos por useFetchGifs", () => {
    const gifs = [
      {
        id: "123",
        url: "https://asd.com/img.jpg",
        title: "hola",
      },
      {
        id: "456",
        url: "https://asd.com/img.jpg",
        title: "asd",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category="messi" />);

    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
