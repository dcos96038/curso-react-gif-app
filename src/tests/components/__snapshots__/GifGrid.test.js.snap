// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid El componente debe renderizar correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    messi
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
